File: problems on numbers/Accept number from user and return the count of digits in between 3 and 7/C++/Main.cpp
//Accept number from user and return the count of digits in between 3 and 7

#include <iostream>
using namespace std;

class DigitsX
{
private:
	
	int iNum ;
public:
	
	DigitsX( int iNum )  
	{
		this -> iNum = iNum;
	}
	
	int countBtwin3and7()
	{
		int iTemp = iNum;
		int iDigit = 0;
		int iCnt = 0;
		
		while ( iTemp != 0 )
		{
			iDigit = iTemp % 10;
			if ( iDigit > 3 && iDigit < 7  )
			{
				iCnt++;
			}
			iTemp = iTemp / 10;
		}
		return iCnt;
	}
	
};

int main()
{
	int iNo = 0; int iCnt = 0;
	cout << " Enter Number " ;
	cin >> iNo;
	
	DigitsX *obj = new DigitsX(iNo);
	
	iCnt = obj -> countBtwin3and7();
	cout << "Count between 3 and 7 is -->" << iCnt << endl ;
	
	delete( obj );
	
	return 0;
}

File: problems on numbers/Accept number from user and return the count of digits in between 3 and 7/Main.java
import java.util.Scanner;

class Main
{
	public static void main ( String args[] )
	{
		System.out.println ( "Enter Number :" );
		Scanner IN = new Scanner ( System.in );
		int iNo = IN.nextInt();
		
		DigitX digit = new DigitX();
		int iRet = digit.countBetween3N7 ( iNo );
		
		System.out.println ( iRet );
	}
}

class DigitX
{
	public int countBetween3N7( int iNo )
	{
		// UPDATOR
		if ( iNo < 0 )
		{
			iNo = -iNo;
		}
		
		int iCnt = 0;
		int iDigit = 0;
		
		while ( iNo != 0 )
		{
			iDigit = iNo % 10 ;
			
			if ( iDigit > 3 && iDigit < 7 )
			{
				iCnt++;
			} 
			
			iNo = iNo / 10;
		}
		return iCnt;
	}
}
